### ISSUE TITLE
`proseWrap` for JSX

### ISSUE THREAD

**Issue by kevincox** [prettier/prettier#16839]
**Prettier 3.3.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAbOMAEAHTBeTACgB0piYAebAPlPPIBkIAnOAW0wEtsBnAVw4ATCKhaYenLAEM2GADSZIUHnDDwYfZpimDunHmE5QA5pjipJAOkwBlPj2wJdPFThYwpi6CoCOfKVioUmD2mHxYAEacEU4C1gAifPqYAGbmyXDMEDwK8GzYoUap6TyYUKoKfKgwzJxgcKUpwZwWEqUqbJZ0lAD0NKQAlADcpCByIBDYMJzeyKBSzFkA7gAKCwg8yCBSqEtSAJ6b4xHMwQDWGDYycAxGcMhNqCrHp2AXMDbYzSbINXxw4w6nF+zH+4zgAA9HLVZLAdgAVTJQBacBoPHbPEASEzoACKfAg8HRTwBIAAVjwITYjMY8QSiUhHpi-IS4CssrwtlIeABacpwQQCsYgGpSFo0gDCEDYbCkXNQqGF2NpcAAgjAatFwmzMrdysTMQALGBsVAAdUNkgaX3qNg2kk4ADdJPstmAXMLHf8AJJQQWwGxgWpTVV+mwwfboA2k7BZFRm07YLblR2ZYVGFTMGDsqTGWXR8ZfZiZrYRKQxRWF2qwM2cQQwQ3IAAcAAZxqw-JxWDm83LGRjSR4IrX642kAAmcb2ODw8ubJB-UnsGKCQWCBhSEz+YxwABiLFlGppXPCEBAAF9z0A)

```sh
--prose-wrap never
```

**Input:**

```jsx
let p = (
	<p>
		Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse porta consequat lacus ut bibendum. Duis felis eros, tempus in felis nec, ultrices facilisis sem.
	</p>
);
```

**Output:**

```jsx
let p = (
	<p>
		Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse porta
		consequat lacus ut bibendum. Duis felis eros, tempus in felis nec, ultrices
		facilisis sem.
	</p>
);
```

**Expected output:**

```jsx
let p = (
	<p>
		Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse porta consequat lacus ut bibendum. Duis felis eros, tempus in felis nec, ultrices facilisis sem.
	</p>
);
```

**Why?**

I expected proseWrap to affect text in JSX. This would allow using soft-wrapping avoid issues with double-wrapping and confusing diffs when adding or removing a few words which causes rewrapping.

At the very least I think the docs should highlight that proseWrap doesn't apply to prose in JSX. Ideally it would be supported (or there would be a separate setting for it for backwards compatibility).

**Comment by mac2000** [comment]
Here is another example that literally makes me crying

<img width="1342" alt="image" src="https://github.com/user-attachments/assets/5fd7e94d-2ccf-4e4c-89af-48b428b59c54" />

**Comment by kevincox** [comment]
I don't think that would qualify as "prose" as there are JSX tags in the line. Maybe you should open a different issue.
